parm/stage/master_gfs.yaml.j2

```diff
@@ -0,0 +1,189 @@
+###################################################################
+# This is the master yaml for the GFS
+#
+# Cycle, member, and RUN settings are set before including each
+# component yaml based on DO switches
+#
+# The included yamls are intended to be of the following structure:
+# key1:
+#     mkdir:
+#         - "COM directory to create"
+#     copy:
+#         - ["source_file", "destination_file"]
+# key2:
+#     mkdir:
+#         - "COM directory to create"
+#     copy:
+#         - ["source_file", "destination_file"]
+#
+# Any number of keys with nested mkdir and copy are permitted
+# Jinja is permitted in this yaml, as long as the keys are:
+# - COMOUT_
+# - DO_ATM, DO_OCN, DO_ICE, etc.
+# For a full list see scripts/exglobal_stage_ic.py
+###################################################################
+
+# Set cycle date variables
+# ------------------------
+{% set half_window = assim_freq // 2 %}
+{% set half_window_begin = (-half_window | string + "H") | to_timedelta %}
+{% set half_window_end = (half_window | string + "H") | to_timedelta %}
+{% if DOIAU and MODE == "cycled" %}
+    {% set model_start_date_current_cycle = current_cycle | add_to_datetime(half_window_begin) %}
+{% else %}
+    {% if REPLAY_ICS %}
+        {% set model_start_date_current_cycle = current_cycle | add_to_datetime(half_window_end) %}
+    {% else %}
+        {% set model_start_date_current_cycle = current_cycle %}
+    {% endif %}
+{% endif %}
+
+{% set current_cycle_YMD = current_cycle | to_YMD %}
+{% set current_cycle_HH = current_cycle | strftime("%H") %}
+{% set previous_cycle_YMD = previous_cycle | to_YMD %}
+{% set previous_cycle_HH = previous_cycle | strftime("%H") %}
+{% set p_prefix = previous_cycle | strftime("%Y%m%d.%H0000") %}
+{% set m_prefix = model_start_date_current_cycle | strftime("%Y%m%d.%H0000") %}
+
+# Determine restart RUN
+# ---------------------
+{% set rRUN = RUN %}
+{% if RUN == "gfs" %}
+    {% set rRUN = "gdas" %}
+{% endif %}
+
+# Set first/last mem for loop
+# ---------------------------
+{% if NMEM_ENS > 0 and RUN == "enkfgdas" %} # Ensemble RUN
```

Comment:
So, the reasoning behind this line...I added the 2nd condition (`RUN == "enkfgdas"`) because when it runs for `RUN=gdas` it was going into this block since `NMEM_ENS>0` is set for the whole shebang (e.g. `NMEM_ENS=2` for the CI tests with enkf). Adding the check of enkfgdas pushes the gdas instance into the other part of the block. Let me know if I didn't answer your question.

Reply:
I see, just wondering if this could be simplified to
```
{% if RUN == "enkfgdas" %} # Ensemble RUN
```
since `NMEM_ENS` _should_ be > 0.  As is, if a user changes `NMEM_ENS` to 0 without rerunning `setup_xml.py`, this will cause the `enkfgdasstage_ic` job to go down the deterministic branch.  An unlikely situation, but one that might be tricky to debug.  But this would be more than a `stage_ic` problem and probably the best way to handle it would be to add a sanity `RUN == enkfgdas && NMEM_ENS <= 0`.

All of that is probably overkill, but seeing the conditional here made me wonder.  I think another issue to add a sanity check script to, say, `jjob-header.sh` would be the right way to handle these kind of edge cases.

Reply:
> if a user changes NMEM_ENS to 0 without rerunning `setup_xml.py`

Ah, I see your point and I like your suggestion. I will adjust this block.

Reply:
I am going to wait to see how the CI tests run on Hera before committing the change. Any new commits will blow away the current approvals and I might as well wait to make sure no further fixes are needed from the CI testing.

Reply:
Sounds good to me.

Reply:
Committed to PR branch @ 13dfad25